src/content/blog/web-framework-performance-report.mdx

```diff
@@ -0,0 +1,132 @@
+
+Most of our frameworks pass this test handily, with over 90% or more of websites passing the assessment. No framework drops below an 80% pass rate on this test. This means that most of the websites tested are responsive to the first user interaction.
+
+<CWVCLSChart />
+
+[Cumulative Layout Shift (CLS)](https://web.dev/cls/) measures visual stability on the page. To pass this assessment, you should reduce unexpected layout shift to near-zero to give your users a reliable visual experience.
+
+CLS is an interesting metric for Google to include as one of the three Core Web Vitals because it isn't strictly related to speed or responsiveness. It's inclusion underscores the importance of looking at more than just performance when it comes to measuring the overall quality of user experiences on the web.
+
+All frameworks scored 50% or higher in this metric. However, it's the youngest frameworks (Astro, SvelteKit and Remix) that score the highest on this metric. All three scored over 75% on the assessment of this metric across all websites tested.
+
+<CWVLCPChart />
+
+[Largest Contentful Paint (LCP)](https://web.dev/lcp/) is the last of the three Core Web Vitals, and arguably the most important when it comes to perceived performance. It measures the point when the page's main content has likely loaded. An LCP of 2.5 seconds or less is required to pass Google's CWV Assessment. Anything slower is considered needing improvement and failing the assessment.
+
+LCP is the hardest of the three metrics to master. Only 52% of all websites tested pass this metric. Of our six frameworks tested, only Astro and SvelteKit beat this average. The rest come in below the average.
+
+# Coming Soon: Interaction to Next Paint (INP)
+
+**Interaction to Next Paint (INP)** is an experimental web vital that assesses overall website responsiveness, similar to First Input Delay (FID). Where the two metrics differ is that INP observes the latency of all interactions a user has made with the page, not just the first one. It also A low INP means the page was consistently able to respond quickly to all—or the vast majority—of user interactions.
+
+While INP is not an official core web vital today, the Chrome team has signaled [their hope](https://www.youtube.com/watch?v=Mizzbsvv8Os&t=378s) to replace FID with INP as a more holistic, accurate measure of responsiveness.
+
+So, how do our frameworks stack up against this new responsiveness metric?
+
+<CWVINPChart />
+
+Most noticible in the chart is that a good INP measurement is overall _much_ harder for our frameworks to achieve than First Input Delay (FID). While every framework tested saw an 80%+ pass rate of FID, no framework was able to see that same 80% pass rate on INP. Astro came closest, at 68.8% passing.
```

Comment:
A general comment that I'll leave here, I guess.

The phrasing "our frameworks" feels a bit out of place. Would prefer something like "target frameworks" or "the frameworks" rather than "our"